app/models/user.rb

```diff
@@ -14,16 +15,42 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  # Check if remembered token is correct password
+  def authenticated? remember_token
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  # Remember an user
+  def remember
+    self.remember_token = User.new_token
```

Comment:
self ở đây là gì e ?

Reply:
Trong objects thì các toán tử gán nếu không dùng self thì ruby sẽ hiểu là tạo local variable ạ. Vì thế ở đây ta phải dùng self để đảm bảo rằng ta đang sử dụng biến remeber_token của object User hiện tại gọi tới hàm remember.